modules/blocks/verify.js

```diff
@@ -801,10 +807,19 @@ Verify.prototype.processBlock = function(block, broadcast, saveBlock, cb) {
 				// * Block and transactions have valid values (signatures, block slots, etc...)
 				// * The check against database state passed (for instance sender has enough LSK, votes are under 101, etc...)
 				// We thus update the database with the transactions values, save the block and tick it.
+				// Also that function set new block as our last block
 				modules.blocks.chain.applyBlock(block, saveBlock, seriesCb);
 			},
-			broadcastBlock(seriesCb) {
-				__private.broadcastBlock(block, broadcast, seriesCb);
+			// Perform next two steps only when 'broadcast' flag is set, it can be:
+			// 'true' if block comes from generation or receiving process
+			// 'false' if block comes from chain synchronisation process
+			updateSystemHeaders(seriesCb) {
+				// Update our own headers: broadhash and height
+				broadcast ? modules.system.update(seriesCb) : seriesCb();
+			},
+			broadcastHeaders(seriesCb) {
+				// Notify all remote peers about our new headers
+				broadcast ? modules.transport.broadcastHeaders(seriesCb) : seriesCb();
```

Comment:
As stated above, if `broadcastHeaders` is moved into `system.update` then we don't need to add it every where.